resotocore/tools/render_dot.py

```diff
@@ -0,0 +1,212 @@
+            type_name = item.get("type")
+            if type_name == "node":
+                uid = value_in_path(item, NodePath.node_id)
+                if uid:
+                    name = value_in_path_get(item, NodePath.reported_name, "n/a")
+                    kind = value_in_path_get(item, NodePath.reported_kind, "n/a")
+                    account = value_in_path_get(item, NodePath.ancestor_account_name, "graph_root")
+                    id = value_in_path_get(item, NodePath.reported_id, "n/a")
+                    parsed_kind = parse_kind(kind)
+                    paired12 = kind_colors.get(parsed_kind, colors[kind])
+                    in_account[account].append(uid)
+                    resource = ResourceDescription(uid, name, id, parsed_kind, kind)
+                    yield render_resource(resource, icon_map, paired12)
+            elif type_name == "edge":
+                from_node = value_in_path(item, NodePath.from_node)
+                to_node = value_in_path(item, NodePath.to_node)
+                if from_node and to_node:
+                    yield f' "{from_node}" -> "{to_node}" '
+        else:
+            raise AttributeError(f"Expect json object but got: {type(item)}: {item}")
+    # All elements in the same account are rendered as dedicated subgraph
+    for account, uids in in_account.items():
+        yield f' subgraph "{account}" {{'
+        for uid in uids:
+            yield f'    "{uid}"'
+        yield " }"
+
+    yield "}"
+
+def ensure_assets():
+
+    if not os.path.exists("Assets"):
+        print("AWS icon assets missing. Downloading assets...")
+        r = requests.get("https://d1.awsstatic.com/webteam/architecture-icons/q1-2022/Asset-Package_01312022.735e45eb7f0891333b7fcce325b0af915fd44766.zip")
+        with open("./Asset-Package.zip", "wb") as f:
+            f.write(r.content)
+        with zipfile.ZipFile("Asset-Package.zip", "r") as zip_ref:
+            zip_ref.extractall("Assets")
+        os.remove("Asset-Package.zip")
+        print("Downloading done.")
+
+def send_analytics():
+    if 'RESOTOCORE_ANALYTICS_OPT_OUT' not in os.environ:
+        client = Client(api_key="n/a", host="https://analytics.some.engineering", flush_interval=0.5, max_retries=3, gzip=True)
+        api_key = requests.get("https://cdn.some.engineering/posthog/public_api_key").text.strip()
+        client.api_key = api_key
+        for consumer in client.consumers:
+            consumer.api_key = api_key
+        run_id = uuid_str()
+        system_id = f"dot-rendering-script"
+        now = utc()
+        client.identify(system_id, {"run_id": run_id, "created_at": now})
+        client.capture(
+            distinct_id=system_id,
+            event="dot-rendering-script-run",
+            properties={"run_id": run_id},  # type: ignore
+            timestamp=now,
+        )    
+
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("json_dump", help="Resoto json dump")
```

Comment:
Just thinking if it would be much easier, if the user could call this script with a query directly?
This script could call the search/graph endpoint and get the resulting json.
wdyt?